controllers/operatorconfigmap_controller.go

```diff
@@ -317,10 +322,35 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	c.log.Info("prometheus rules deployed", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
 
+	// deletion phase
+	if !instance.GetDeletionTimestamp().IsZero() {
+		err = c.deletionPhase(&req)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
 	return ctrl.Result{}, nil
 }
 
-func (c *ClusterVersionReconciler) createOrUpdate(obj client.Object, f controllerutil.MutateFn) error {
+func (c *OperatorConfigMapReconciler) deletionPhase(req *ctrl.Request) error {
+	csiAddonsNode := &csiaddonsv1alpha1.CSIAddonsNode{}
+	err := c.Client.Get(c.ctx, req.NamespacedName, csiAddonsNode)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted before reconcile request.
+			c.log.Info("CSIAddonsNode resource not found")
+			return nil
+		}
+		return err
+	}
+	csiAddonsNode.Finalizers = nil
+	if err := c.Client.Update(c.ctx, csiAddonsNode); err != nil {
+		c.log.Error(err, "Failed to remove finalizer from csiAddonsNode")
+		return err
+	}
+	return nil
```

Comment:
This is not the established way to remove finalizers. 
Please take a look at this code snippet from another repo we are working on: https://github.com/red-hat-storage/ocs-osd-deployer/blob/main/controllers/managedocs_controller.go#L508-L513

Reply:
AFAIC this can lead to unexpected errors as we are not the ones creating the `CSIAddonsNode`, I think this is happening due to the way the uninstall is happening but this is not the right component to fix this.

Reply:
@Madhu-1 where should the fix go?

Reply:
it should go to the component `csi-addons` but that needs to be discussed with the team first. Do we have a BZ for it?

Reply:
This entire PR is a bug fix. See the bug number int the PR title

Reply:
@bernerhat I agree with @Madhu-1 that the code that adds the finalizer should remove the finalizer. 
Are we even sure that with our ownership changes, there is still an issue with the removal of these finalizers? Are we making sure to add resource ownership to all the resources this reconciler reconciles?

Reply:
I agree as well, it makes a lot more sense, I had some issues trying out the code so i did not make sure yet if the ownership changes solves the finalizer issue as well.
I believe that we do add ownership to all other resources reconciled here.